catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswSubscriptionEndpoint.java

```diff
@@ -280,6 +308,88 @@ public Response createOrUpdateSubscription(GetRecordsType request, String reques
         return createAcknowledgment(request);
     }
 
+    @Override
+    @POST
+    @Path("/event")
+    @Consumes({MediaType.TEXT_XML, MediaType.APPLICATION_XML})
+    @Produces({MediaType.TEXT_XML, MediaType.APPLICATION_XML})
+    public Response createEvent(GetRecordsResponseType recordsResponse) throws CswException {
+        if (!METACARD_SCHEMA.equals(recordsResponse.getSearchResults()
+                .getRecordSchema())) {
+            throw new CswException("Only " + METACARD_SCHEMA + " is supported for ingest events");
+        }
+        List<Metacard> metacards = getMetacards(recordsResponse);
+        eventProcessor.notifyCreated(metacards.get(0));
+        return Response.ok()
+                .build();
+
+    }
+
+    @Override
+    @PUT
+    @Path("/event")
+    @Consumes({MediaType.TEXT_XML, MediaType.APPLICATION_XML})
+    @Produces({MediaType.TEXT_XML, MediaType.APPLICATION_XML})
+    public Response updateEvent(GetRecordsResponseType recordsResponse) throws CswException {
+        if (!METACARD_SCHEMA.equals(recordsResponse.getSearchResults()
+                .getRecordSchema())) {
+            throw new CswException("Only " + METACARD_SCHEMA + " is supported for update events");
+        }
+        List<Metacard> metacards = getMetacards(recordsResponse);
+        eventProcessor.notifyUpdated(metacards.get(0), null);
+        return Response.ok()
+                .build();
+
+    }
+
+    @Override
+    @DELETE
+    @Path("/event")
+    @Consumes({MediaType.TEXT_XML, MediaType.APPLICATION_XML})
+    @Produces({MediaType.TEXT_XML, MediaType.APPLICATION_XML})
+    public Response deleteEvent(GetRecordsResponseType recordsResponse) throws CswException {
+        if (!METACARD_SCHEMA.equals(recordsResponse.getSearchResults()
```

Comment:
Wrap in a method...pass just the one word difference to the method and use `String.format` to compose your exception message